materialize/resources/resource_secret.go

```diff
@@ -71,7 +71,7 @@ func newSecretBuilder(secretName, schemaName, databaseName string) *SecretBuilde
 }
 
 func (b *SecretBuilder) Create(value string) string {
-	return fmt.Sprintf(`CREATE SECRET %s.%s.%s AS %s;`, b.databaseName, b.schemaName, b.secretName, value)
+	return fmt.Sprintf(`CREATE SECRET %s.%s.%s AS '%s';`, b.databaseName, b.schemaName, b.secretName, value)
```

Comment:
Eep, this isn't safe! It will fail for secret values that include single quotes. 

The safe way to do this is to use the `pq.QuoteLiteral` function:

https://github.com/lib/pq/blob/d5affd5073b06f745459768de35356df2e5fd91d/conn.go#L1631-L1681

Reply:
Good catch! I've updated this and added a couple more tests for this as well